src/Http/Controllers/UserController.php

```diff
@@ -70,53 +67,33 @@ public function register(Request $request)
             $user->roles($roles);
         }
 
-        // TODO: Registering event
+        if (UserRegistering::dispatch($user) === false) {
+            return $this->userRegistrationSuccess(true);
+        }
```

Comment:
If we're aiming to mimic the Form submission behavior (which, you're right, we should) then we should borrow the try/catch style with a SilentFormFailureException from here:

https://github.com/statamic/cms/blob/a3505b787c003cdd3e22db37013305a93391ac4c/src/Http/Controllers/FormController.php#L57-L62

You can leave the existing validation where it is, but we'd just need to let your listener throw a ValidationException as described here: https://statamic.dev/extending/events#formsubmitted

Reply:
Ya I had that the first time I did this, then changed my mind...lol